app/src/main/java/com/mk/tasky/core/presentation/TaskyTextField.kt

```diff
@@ -0,0 +1,157 @@
+package com.mk.tasky.core.presentation
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Check
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mk.tasky.ui.theme.*
+
+@Composable
+fun TaskyTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String = "",
+    showError: Boolean = false,
+    maxLines: Int = 1,
+    keyboardType: KeyboardType = KeyboardType.Text,
+    imeAction: ImeAction = ImeAction.Done,
+    trailingIcon: ImageVector? = null,
+    trailingIconColor: Color? = null,
+    onIconClick: () -> Unit = {},
+    isPassword: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    var isPasswordHidden by remember {
+        mutableStateOf(true)
+    }
```

Comment:
This would not survive a config change, prefer to put all states in your VM and provide them as a parameter to the composable. Using `rememberSaveable` would survive a config change and would be debatable, but I'd value the consistency of having a single source where you manage the state (which is in the VM). This would rather be an XML way of doing things.

Reply:
Perfect. Will remove the state!